packages/platform-core/src/caller_directory_extractor.ts

```diff
@@ -0,0 +1,42 @@
+import path from 'path';
+import * as os from 'os';
+import { FilePathExtractor } from './extract_file_path_from_stack_trace_line';
+
+/**
+ * Extracts the path of the caller of the code that generated the input stack trace.
+ * In other words, extracts the path from the _second_ entry in the stack trace
+ * (the first entry being the location where the stack trace was created and the second entry being the location that called the code that generated the stack trace)
+ */
+export class CallerDirectoryExtractor {
+  /**
+   * Creates caller directory extractor.
+   */
+  constructor(private readonly stackTrace: string | undefined) {}
+
+  extract = () => {
+    let stackTrace = this.stackTrace;
+    const unresolvedImportLocationError = new Error(
+      'Could not determine import path to construct absolute code path from relative path. Consider using an absolute path instead.'
+    );
+    if (!stackTrace) {
+      throw unresolvedImportLocationError;
+    }
+    // normalize EOL to \n so that parsing is consistent across platforms
+    stackTrace = stackTrace.replaceAll(os.EOL, '\n');
+    const stacktraceLines =
+      stackTrace
+        .split('\n')
+        .map((line) => line.trim())
+        .filter((line) => line.startsWith('at')) || [];
+    if (stacktraceLines.length < 2) {
+      throw unresolvedImportLocationError;
+    }
+    const stackTraceImportLine = stacktraceLines[1]; // the first entry is the file where the error was initialized (our code). The second entry is where the customer called our code which is what we are interested in
+
+    const filePath = new FilePathExtractor(stackTraceImportLine).extract();
+    if (filePath) {
+      return path.dirname(filePath);
+    }
+    throw unresolvedImportLocationError;
+  };
```

Comment:
This logic is moved as is from `packages/backend-function/src/get_caller_directory.ts`

Reply:
Curious why we switched this to become a class now, could we have kept this as a function?

Reply:
So that it looks consistent next to `FilePathExtractor` in https://github.com/aws-amplify/amplify-backend/blob/main/packages/platform-core/API.md .